### PR TITLE
adjust status on container start failure

### PR DESCRIPTION
**Pull request by runcom**
@mrunalp @sameo @Random-Liu PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
second commit just uses `pools` from docker pkg to do `io.Copy` operation to save some memory when doing huge streams

File: server/container_start.go [hunk 1]
```diff
@@ -15,13 +16,23 @@ func (s *Server) StartContainer(ctx context.Context, req *pb.StartContainerReque
 	if err != nil {
 		return nil, err
 	}
+	state := s.runtime.ContainerStatus(c)
+	if state.Status != oci.ContainerStateCreated {
+		return nil, fmt.Errorf("container %s is not in created state: %s", c.ID(), state.Status)
+	}
+
+	defer func() {
```
Comment (sameo) [inline server/container_start.go]: Could we explain why this needs to be deferred ?

Comment (runcom) [inline server/container_start.go]: added, PTAL

Comment (sameo) [inline server/container_start.go]: Grazie.

Comment (runcom) [inline server/container_start.go]: De rien!

Comment (runcom) [inline server/container_start.go]: Actually I amended the comment to the wrong commit lol I'll fix it.

**Comment by sameo** [comment]
> second commit just uses pools from docker pkg to do io.Copy operation to save some memory when doing huge streams

@runcom Could we please add something along those lines to the 2nd commit's message ? Me being unfamiliar with docker's pools package, I had to go and look at the (completely undocumented) code to understand what it's doing.

**Comment by runcom** [comment]
@sameo yup I'll do that as well

**Comment by runcom** [comment]
should be good to go now @sameo @mrunalp

**Comment by sameo** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**